.github/workflows/build.yml

```diff
@@ -0,0 +1,165 @@
+        node: ["14"]
+        rails_version: ["6.1", "7.0", "7.2", "edge"] # "rubocop", "i18n-tasks"
+        # db: [sqlite3, postgres, mysql]
+        db: [sqlite3, postgres]
+        exclude:
+          # Not supported versions
+          - ruby: 2.7.8
+            rails_version: 7.2
+          - ruby: 3.0.7
+            rails_version: 7.2
+          - ruby: 2.7.8
+            rails_version: edge
+          - ruby: 3.0.7
+            rails_version: edge
+
+          # Logger Issues
+          - ruby: 3.3.4
+            rails_version: edge
+          - ruby: 3.3.4
+            rails_version: 7.2
+          - ruby: 3.2.4
+            rails_version: edge
+          - ruby: 3.2.4
+            rails_version: 7.2
+          - ruby: 3.1.6
+            rails_version: edge
+          - ruby: 3.1.6
+            rails_version: 7.2
+
+          # Psych issues
+          - ruby: 3.3.4
+            rails_version: 6.1
+          - ruby: 3.2.4
+            rails_version: 6.1
+          - ruby: 3.1.6
+            rails_version: 6.1
+
+
+        # include:
+        #   - ruby: 2.7
+        #     rails: rubocop
+        #     db: null
+        #   - ruby: 2.7
+        #     rails: i18n-tasks
+        #     db: null
+
+    env:
+      RAILS_ENV: test
+      COVERAGE: 1
+      FERRUM_DEFAULT_TIMEOUT: 20
+      BUNDLE_GEMFILE: "spec/gemfiles/rails_${{ matrix.rails_version }}.gemfile"
+      DB_USERNAME: thredded
+      DB_PASSWORD: thredded
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set DB adapter
+      run: |
+        export DB=${{ matrix.db }}
+
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby }}
+        bundler-cache: true # install gems and cache
+
+    - uses: browser-actions/setup-chrome@v1
+
+    - uses: actions/setup-node@v4
+      with:
+        node-version: ${{ matrix.node }}
+
+    - name: Install Node Packages via Yarn
+      run: |
+        if [[ $BUNDLE_GEMFILE != "${PWD}/spec/gemfiles/rails_5_2.gemfile" ]] && [[ -z $THREDDED_TESTAPP_SPROCKETS_JS ]]; then
+          cd spec/dummy && yarn && cd -
+        fi
+
+    - name: "Run Tests: test:nojs"
+      run: |
+        SIMPLECOV_NAME="test:nojs" bundle exec rspec --force-color --format d --tag=~js --tag=~i18n_tasks
+
+    - name: "Run Tests: test:js"
+      run: |
+        SIMPLECOV_NAME="test:js" bundle exec rspec --force-color --format d --tag=js
+
+    - name: "Run Tests: spec/migrations"
+      run: |
+        MIGRATION_SPEC=1 SIMPLECOV_NAME="test:migrations" bundle exec rspec spec/migration --force-color --format d
+
+    - name: "Run Tests: spec/configuration/run_all"
+      run: |
+        spec/configuration/run_all
+
+    - name: "Run Tests: run_spec_repeatedly"
+      run: |
+        bin/run_specs_repeatedly --tag=threaded_render
+
+    # - name: Test & publish code coverage
+    #   uses: paambaati/codeclimate-action@v8.0.0
+    #   env:
+    #     CC_TEST_REPORTER_ID: <code_climate_reporter_id>
```

Comment:
If we want to get CodeClimate working again, then we'll need to determine the CC_TEST_REPORTER_ID: See https://docs.codeclimate.com/docs/finding-your-test-coverage-token